runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkPipelineRunner.java

```diff
@@ -0,0 +1,212 @@
+    try {
+      parser.parseArgument(args);
+    } catch (CmdLineException e) {
+      LOG.error("Unable to parse command line arguments.", e);
+      parser.printUsage(System.err);
+      throw new IllegalArgumentException("Unable to parse command line arguments.", e);
+    }
+    return configuration;
+  }
+
+  private void startJobService() throws Exception {
+    final PrintStream oldErr = System.err;
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    PrintStream newErr = new PrintStream(baos);
+    try {
+      System.setErr(newErr);
+      driver =
+          FlinkJobServerDriver.fromParams(
+              new String[] {"--job-port=" + jobPort, "--artifact-port=0", "--expansion-port=0"});
+      driverThread = new Thread(driver);
+      driverThread.start();
+      boolean success = false;
+
+      // TODO: check for job service ready
+      Thread.sleep(5000);
+      success = true;
+
+      while (!success) {
+        newErr.flush();
+        String output = baos.toString(Charsets.UTF_8.name());
+        if (output.contains("JobService started on localhost:")
+            && output.contains("ArtifactStagingService started on localhost:")
+            && output.contains("ExpansionService started on localhost:")) {
+          success = true;
+        } else {
+          Thread.sleep(100);
+        }
+      }
+
+      if (!driverThread.isAlive()) {
+        throw new IllegalStateException("Job service thread is not alive");
+      }
+    } finally {
+      System.setErr(oldErr);
+    }
+  }
+
+  private void runDriverProgram() throws Exception {
+    ProcessManager processManager = ProcessManager.create();
+    String executable = "bash";
+    List<String> args =
+        ImmutableList.of("-c", String.format("exec %s " + DRIVER_CMD_FLAGS, driverCmd, jobPort));
+    String processId = "client1";
+
+    Duration timeout = Duration.ofSeconds(30);
+    Deadline deadline = Deadline.fromNow(timeout);
+    try {
+      final Process driverProcess =
+          processManager
+              .startProcess(processId, executable, args, System.getenv())
+              .getUnderlyingProcess();
+      LOG.info("Started driver program");
+      while (driverProcess.isAlive() && deadline.hasTimeLeft()) {
+        try {
+          Thread.sleep(1000);
+        } catch (InterruptedException interruptEx) {
+          Thread.currentThread().interrupt();
+          throw new RuntimeException(interruptEx);
+        }
+      }
+      if (driverProcess.isAlive()) {
+        String msg =
+            String.format(
+                "Timeout of %s waiting for command '%s' to submit the job.", deadline, args);
+        throw new TimeoutException(msg);
+      } else {
+        if (driverProcess.exitValue() == 0) {
+          // convey to the environment that the job was successfully constructed
+          throw new OptimizerPlanEnvironment.ProgramAbortException();
+        }
+        throw new RuntimeException("Driver program failed.");
```

Comment:
```suggestion
        throw new RuntimeException("Driver program failed with exit value " + driverProcess.exitValue());
```

Printing the error output could also be useful.